lib/containers/urls.pm

```diff
@@ -51,64 +51,220 @@ sub get_opensuse_registry_prefix {
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x']
+        }
+    },
+    opensuse => {
+        Tumbleweed => {
+            released => sub { 'registry.opensuse.org/opensuse/tumbleweed' },
+            totest   => sub {
+                'registry.opensuse.org/' . get_opensuse_registry_prefix . 'opensuse/tumbleweed';
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x', 'arm']
+        },
+        '15.0' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.0' },
+            totest   => sub {
+                'registry.opensuse.org/opensuse/leap/15.0/images/totest/images/opensuse/leap:15.0';
+            },
+            available_arch => ['x86_64']
+        },
+        '15.1' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.1' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.1/images/totest/containers/opensuse/leap:15.1';
+                } elsif ($arch eq 'arm') {
+                    'registry.opensuse.org/opensuse/leap/15.1/arm/images/totest/containers/opensuse/leap:15.1';
+                } else {
+                    '';
+                }
+
+            },
+            available_arch => ['x86_64', 'arm']
+        },
+        '15.2' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.2' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.2/images/totest/containers/opensuse/leap:15.2';
+                } elsif ($arch eq 'arm') {
+                    'registry.opensuse.org/opensuse/leap/15.2/arm/images/totest/containers/opensuse/leap:15.2';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'arm']
+        },
+        '15.3' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.3' },
+            totest   => sub {
+                my $arch = shift;
+                if (grep { $_ eq $arch } qw/x86_64 aarch64 ppc64le s390x/) {
+                    'registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3';
+                } elsif ($arch eq 'arm') {
+                    'registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x', 'arm']
+        }
+    },
+    'sle-micro' => {
+        '15' => {
+            released       => sub { 'registry.suse.com/suse/sle15:15.0' },
+            totest         => sub { },
+            available_arch => ['x86_64', 'ppc64le', 's390x']
```

Comment:
That we do not have `aarch64` is new in the refactored code, Is it expected?

Reply:
thats a good point. 
suse/containers/sle-server/15/containers/suse/sle15 with tag `15.0`, `arm64` is missing. Although if you check `latest` tag the `arm64` seems available. `latest` is tricky because it might be different build than `15.0`. Also the repo in obs, there are images built against aarch64 but not for `15_0`. 
So imo `aarch` shouldnt be in this list.

Reply:
ok

Reply:
That's right, as in previous comment, 15(SP0) aarch64 doesn't exist.